contents/docs/architecture/ingestion-pipeline.mdx

```diff
@@ -0,0 +1,135 @@
+
+#### `$identify` or `$alias` event
+
+|                                                                                                                                                 |                                                                                                              |
+| ----------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------ |
+| If we have never seen this `distinct_id` before                                                                                                 | Create a new Person and add a mapping in PostgreSQL to associate this `distinct_id` with the new `person_id` |
+| If there is a single Person already associated with this user                                                                                   | Create a new mapping to associate this `distinct_id` with the already existing `person_id`                   |
+| If there is already a Person associated with this `distinct_id` _and_ we have already created a second Person that is associated with this user | We will [merge these two people](#merging-two-persons) and associate all future events with the first Person |
+
+##### Merging two Persons
+
+In the third scenario, where we have inadvertently created two Persons for the same user, we will need to merge them. In the case of an `$identify` call, we will create a mapping from the `distinct_id` of the anonymous user to the non-anonymous user.
+If there are any conflicts when merging Person properties for these two users, the values from the non-anonymous user will take precedence.
+We choose to prioritize the history of the non-anonymous user, as it is far more likely that this user will have a history of previous events associated with them that we want to preserve.
+
+In the case of an `$alias` call, we will merge the second user into the first depending on the order these users were provided to the call.
+
+##### Consequences of merging
+
+This scenario typically isn't ideal, as merging two users will only affect events that are sent in the future.
+As a result, we will be left with two separate Persons that shoulb be associated with the same user, but instead will be treated as entirely unique users.
+The only way to fully 'merge' these two people would be to go back and rewrite the `person_id` field on past events, which is not practical or performant.
+
+Avoiding this scenario is the primary goal of the event buffer, as by buffering events from users we have never seen before, we can avoid creating a duplicate Person if it turns out they have been identified or logged in shortly after sending this first event.
```

Comment:
I mentioned above: two cases - events arriving out of order or the sign-up flow contains backend events (which would use the identified distinct_id) which could arrive before `identify` is called on the frontend (events before identify would use the anon_distinct_id). So it's like if shortly after seeing an event with the identified distinct_id we see identify call we'll be able to have the single person_id for all the events